### PR TITLE
Fixed build linking error osx v11.0.1

### PR DESCRIPTION
**Pull request by blexaer**
See issue #179

### Description
Added version checking to linking in cmake

### How Has This Been Tested?
Tested as working by changing required version accordingly.

### CODE REVIEW

File: client/CMakeLists.txt [hunk 1]
```diff
@@ -51,13 +51,21 @@ set(trillek-client_SOURCES ${trillek-client_SOURCES} PARENT_SCOPE) # so tests ca
 
 # macOS needs to link libobjc
 set(OSX_LIBRARIES "")
-if (APPLE)
+
+
+IF (APPLE)
+EXEC_PROGRAM(uname ARGS -v OUTPUT_VARIABLE DARWIN_VERSION)
```
Comment (adam4813) [inline client/CMakeLists.txt:57]: Just a small nit-pick, but can you indent these some to match the rest of the file? Also there appears to be an extra newline on line 54.

Comment (blexaer) [reply client/CMakeLists.txt:57]: Sure, on it.

**Review by adam4813 — APPROVED** [review]

File: client/CMakeLists.txt (hunk 1 above)
Comment (adam4813) [reply client/CMakeLists.txt]: Looks good

File: README.md [hunk 2]
```diff
@@ -18,16 +18,16 @@ TEC requires cmake 3.9 and a few libraries GLFW3, GLM, ASIO, Protobuf, GLEW, Lua
 	`./bootstrap-vcpkg.bat`
 	**OPTIONAL** `./vcpkg integrate install`
 	`./vcpkg install asio bullet3 glew glfw3 glm lua openal-soft protobuf zlib spdlog imgui sol2`
-#### MacOS (NOT TESTED) 
-	 `sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /` if `bootstrap-vcpkg` fails see [here](https://donatstudios.com/MojaveMissingHeaderFiles) for more help
-	 `./bootstrap-vcpkg.sh`
-	 **OPTIONAL** `./vcpkg integrate install`
-	 `./vcpkg install asio bullet3 glew glfw3 glm lua openal-soft protobuf zlib spdlog imgui sol2`
+#### MacOS
+	Prior to 11.0.1, first run (NOT TESTED): `sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /` if `bootstrap-vcpkg` fails see [here](https://donatstudios.com/MojaveMissingHeaderFiles) for more help
```
Comment (adam4813) [inline README.md]: You left some ticks on this line, was this intentional?

File: README.md [hunk 3]
```diff
@@ -18,16 +18,16 @@ TEC requires cmake 3.9 and a few libraries GLFW3, GLM, ASIO, Protobuf, GLEW, Lua
...
+	Prior to 11.0.1, first run (NOT TESTED): `sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /` if `bootstrap-vcpkg` fails see [here](https://donatstudios.com/MojaveMissingHeaderFiles) for more help 
+	./bootstrap-vcpkg.sh
+	**OPTIONAL** ./vcpkg integrate install
+	./vcpkg install asio bullet3 glew glfw3 glm lua openal-soft protobuf zlib spdlog imgui sol2
 #### Linux
-	 `apt-get install libgl1-mesa-dev xorg-dev libglu1-mesa-dev libxinerama-dev libxcursor-dev`
-	 `./bootstrap-vcpkg.sh`
-	 **OPTIONAL** `./vcpkg integrate install`
-	 `./vcpkg install asio bullet3 glew glfw3 glm lua openal-soft protobuf zlib spdlog imgui sol2`
+	`apt-get install libgl1-mesa-dev xorg-dev libglu1-mesa-dev libxinerama-dev libxcursor-dev`
```
Comment (adam4813) [inline README.md]: These few lines still have ticks, as this intentional?

**Review by adam4813 — APPROVED** [review]

**Review by adam4813 — APPROVED** [review]

**Pull request merged**